### PR TITLE
Add getSchema using group name, schema name, and version

### LINKED ISSUE #30822: Provide a function for fetching schema from Schema Registry using group name, schema name and schema version
https://github.com/Azure/azure-sdk-for-python/pull/26055

### PR DESCRIPTION
**Pull request by conniey**
# Description

- Adds `getSchema(groupName, schemaName, version)
- Adds `getVersion` to SchemaProperties.

Fixes #30822 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java [hunk 1]
```diff
@@ -182,12 +188,33 @@ Mono<Response<SchemaProperties>> registerSchemaWithResponse(String groupName, St
...
     }
 
+    /**
+     * Gets the schema properties of the schema associated with the group name, schema name, and schema version.
+     *
+     * @param groupName Group name for the schema
+     * @param schemaName Name of the schema
+     * @param schemaVersion Version of schema
+     *
+     * @return The {@link SchemaProperties} matching the parameters.
```
Comment (anuchandy) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java:205]: nit; `SchemaProperties` ->  `SchemaRegistrySchema`

Comment (anuchandy) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java:205]: same return type doc for the related `getSchemaWithResponse` APIs

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java [hunk 2]
```diff
@@ -338,7 +432,30 @@ private static Throwable remapError(ErrorException error) {
...
+            String str;
+
+            while ((str = reader.readLine()) != null) {
+                builder.append(str);
+            }
+
+        } catch (IOException exception) {
+            throw new UncheckedIOException("Error occurred while deserializing schemaContent.", exception);
+        }
+
```
Comment (anuchandy) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java:458]: Is it possible to use `readAllBytes()` API?

```java
        try {
            return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
        } catch (IOException exception) {
            throw new UncheckedIOException("Error occurred while deserializing schemaContent.", exception);
        }
```

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/models/SchemaProperties.java [hunk 3]
```diff
@@ -90,4 +93,13 @@ public String getGroupName() {
...
         return name;
     }
+
+    /**
+     * Gets the version of the schema.
+     *
+     * @return The version of the schema.
+     */
+    public int getVersion() {
+        return version;
```
Comment (anuchandy) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/models/SchemaProperties.java:103]: nit; Is it helpful to document that, if no version exists then returns -1?

**Review by anuchandy — APPROVED** [review]
Thanks Connie! LGTM. (I skipped all generated files and looked only the convenience layer files)

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java (hunk 2 above)
Comment (conniey) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java:458]: Ah yeah. This was introduced in JDK 9, so I didn't use it.

Comment (anuchandy) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java:458]: I see, thanks!

**Pull request merged**